### ISSUE TITLE
differences in log, route & trip file arising from teleports

### ISSUE THREAD

**Issue by ambuehll** [eclipse/sumo#10597]
I have multiple vehicles which teleport (which is not the problem). The "traces" they leave in the route file are my problem. Here is the route output of vehicle `H26451`:

```
vehicle id="H26451" depart="30235.00" departLane="free" departSpeed="8.2079" fromTaz="1544" toTaz="1606" speedFactor="1.0944" arrival="30756.00" routeLength="1675.78">
        <route edges="-75528935#1 75528934#0 75528930#0 75528930#1 -548134627#2 -218931349#5 -5183081#1 534546682#0 533775756#0 549020386#0 549020386#1 534851512#0 147636160#0 11916106#0 11916106#1 722646198 623570426#0 208809259#1 -147692716#2" 
            exitTimes="30261.00     30274.00 30276.00   30287.00    30305.00        30330.00    30640.00    30710.00   30718.00    30732.00    30756.00         -1          -1          -1         -1       -1         -1           -1          -1"/>
    </vehicle>
```
trip output:
```
<tripinfo id="H26451" depart="30235.00" departLane="" departPos="0.00" departSpeed="8.21" departDelay="0.95" arrival="30756.00" arrivalLane="" arrivalPos="197.58" arrivalSpeed="8.21" duration="521.00" routeLength="1028.12" waitingTime="301.81" waitingCount="0" stopTime="0.00" timeLoss="328.68" rerouteNo="1" devices="vehroute_H26451 tripinfo_H26451 routing_H26451" vType="DEFAULT_VEHTYPE" speedFactor="1.09" vaporized=""/>
```

and the log for the same vehicle:

```
Warning: Teleporting vehicle 'H26451'; waited too long, from edge '-5183081#1':0, time=30639.63.
Warning: Vehicle 'H26451' ends teleporting on edge '623570426#0':0, time=30709.47.
```

While the vehicle starts being teleported at 30639 from edge -5183081#1, the route file still receives entries for the next 4 edges (534546682#0 533775756#0 549020386#0 549020386#1). Note it exits at 30756 the fourth edge 549020386#1.  Even though according to the log the vehicle arrived at the much later edge of the route BEFORE arriving at the previous edges mentioned above 30709.47.

As always, thanks for looking into this. 
and, as always: [files](https://drive.google.com/drive/folders/13IBNfyGxQ3TMrPBG41eIRJCiAFJbp2mZ?usp=sharing)

sumo Version v1_12_0+1915-27c190f6120
Build features: Darwin-21.4.0 arm64 Clang 13.1.6.13160021 Release FMI Proj GUI SWIG

**Comment by namdre** [comment]
The example cannot be run because the files "input/additional.add.xml,taz.xml" are missing.

**Comment by ambuehll** [comment]
Sorry! updated the files accordingly. Might also suffer from reproducibiliy issues from threads>0.

**Comment by namdre** [comment]
Yes. Your scenario is subject to #10292 and the vehicle in question does not teleport reliably. 
It would be quite helpful if you could provide an failing example that omits either one of the options **device.rerouting.threads** or **weights.random-factor**.

**Comment by ambuehll** [comment]
Took me a while to find one :)
only posting the last entries of the route file:
```
<vehicle id="F51420" depart="24591.00" departLane="free" departSpeed="4.6399" fromTaz="820" toTaz="1539" speedFactor="0.7520" arrival="25925.00" routeLength="6178.47">
    <route edges="... -541012856 -541012867#2 -541012857 -541012858 -541012895#3 -541012895#2 -541012868#1 5075635 208800979#1 24701208#0" 
    exitTimes=" ...     25414.00     25722.00   25752.00   25766.00     25905.00     25925.00         -1        -1        -1          -1"/>
</vehicle>
```

```
Warning: Teleporting vehicle 'F51420'; waited too long, from edge '-541012867#2':0, time=25721.68.
Warning: Vehicle 'F51420' ends teleporting on edge '5075635':0, time=25751.46.
```

Here edge 5075635 has an exit time of -1 even though according to the log, the vehicle arrived at 25751.46. The time 25751.46 (rounded) however is written for edge -541012857, which is located 5 edges prior to the teleport destination 5075635.

In my opinion there is some mixing up going on for the four last edges `(-541012868#1 5075635 208800979#1 24701208#0)` vs the teleported edges `(-541012857 -541012858 -541012895#3 -541012895#2)`.

files updated in a new folder "singlethread".

**Comment by namdre** [comment]
I ran the singlethread scenario with version 27c190f6120 (as listed above) and there is no teleport by vehicle 'F51420'. 
If you have a script (or some other simple way) to find inconsistent exitTimes in vehroute-output, I may still be able to use this scenario...

**Comment by ambuehll** [comment]
Hmm. That's weird, I double checked and ran it again twice, and got identical results with sumo Version v1_12_0+1915-27c190f6120. I'll update to the current build, and run it again. Unfortunately, I don't have a script for the detection. A student found this and I am just the "forwarder". I'll check with her as well.

**Comment by ambuehll** [comment]
Double checked with current build v1_12_0+2420-9cd1df78d9a
Build features: Darwin-21.4.0 arm64 Clang 13.1.6.13160021 Release FMI Proj GUI SWIG

Still the same output as above (F51420). I am happy to help but I am not really sure how to. Maybe, you could upload your output to the google drive? I'll see if I find some other case.

**Comment by namdre** [comment]
https://drive.google.com/file/d/1v17zLrIPJ_isHT7XWyb1cQASvPjkxEL7/view?usp=sharing

**Comment by ambuehll** [comment]
Thanks. found one in your output files: H26451

```
<route edges="-75528935#1 75528934#0 75528930#0 75528930#1 -548134627#2 -218931349#5 -5183081#1 534546682#0 533775756#0 549020386#0 549020386#1 534851512#0 147636160#0 11916106#0 11916106#1 722646198 623570426#0 208809259#1 -147692716#2" 
exitTimes="30261.00         30274.00 30276.00    30287.00       30305.00    30330.00    30640.00   30710.00     30718.00   30732.00     30756.00         -1          -1        -1         -1        -1          -1          -1          -1"/>
```
Warning: Teleporting vehicle 'H26451'; waited too long, from edge '-5183081#1':0, time=30639.63.
Warning: Vehicle 'H26451' ends teleporting on edge '623570426#0':0, time=30709.47.


exit time for 623570426 is -1 in route file, whereas the log says teleport at 30710 (rounded).  30710 is the exittime of 534546682, way ahead of 623570426.
